src/coreclr/jit/llvm.cpp

```diff
@@ -1654,14 +1630,190 @@ void startImportingNode(llvm::IRBuilder<>& builder)
     }
 }
 
+
+Llvm::Llvm(Compiler* pCompiler)
+{
+    _compiler = pCompiler;
+    _info = pCompiler->info;
+
+    _function = nullptr;
+    _requiresShadowStackAddSubsitution = false;
+    _sigInfoIsValid = false;
+}
+
+void Llvm::llvmShutdown()
+{
+    if (_diBuilder != nullptr)
+    {
+        emitDebugMetadata(_llvmContext);
+    }
+#if DEBUG
+    if (_outputFileName == nullptr) return; // nothing generated
+    std::error_code ec;
+    char* txtFileName = (char*)malloc(strlen(_outputFileName) + 2); // .txt is longer than .bc
+    strcpy(txtFileName, _outputFileName);
+    strcpy(txtFileName + strlen(_outputFileName) - 2, "txt");
+    llvm::raw_fd_ostream textOutputStream(txtFileName, ec);
+    _module->print(textOutputStream, (llvm::AssemblyAnnotationWriter*)NULL);
+    free(txtFileName);
+#endif //DEBUG
+    llvm::raw_fd_ostream OS(_outputFileName, ec);
+    llvm::WriteBitcodeToFile(*_module, OS);
+    delete _module;
+    //    Module.Verify(LLVMVerifierFailureAction.LLVMAbortProcessAction);
+}
+
+GenTree* createAddNodeForShadowStackLocal(LclVarDsc* varDsc, unsigned shadowStackLclNum, GenTreeIntCon*& offset, GenTreeLclVar*& shadowStackVar)
+{
+    // TODO-LLVM: if the offset == 0, just GT_STOREIND at the shadowStack
+    offset = _compiler->gtNewOneConNode(var_types::TYP_INT)->AsIntCon();
+    offset->SetIconValue(varDsc->GetStackOffset());
+    shadowStackVar = _compiler->gtNewLclvNode(shadowStackLclNum, var_types::TYP_REF);
+    return _compiler->gtNewOperNode(genTreeOps::GT_ADD, var_types::TYP_REF, shadowStackVar, offset);
+}
+
+void insertAddWithOperands(LIR::Range& lirRange, GenTree* node, GenTree* addNode, GenTreeIntCon* offset, GenTreeLclVar* shadowStackVar)
+{
+    lirRange.InsertBefore(node, offset);
+    lirRange.InsertAfter(offset, shadowStackVar);
+    lirRange.InsertAfter(shadowStackVar, addNode);
+}
+
+void Llvm::ConvertShadowStackLocals()
+{
+    unsigned shadowStackLclNum = _compiler->lvaGrabTemp(false, "shadowstack"); // TODO-LLVM: create a new local as a temp, is this right?  Copied from lower.cpp
```

Comment:
> is this right

Yep.
```suggestion
    unsigned shadowStackLclNum = _compiler->lvaGrabTemp(true DEBUGARG("shadowstack"));
```
`true` because it "has long lifetime", though it doesn't actually matter (this was used a very long time ago by Pre-RyuJit "legacy" backend as a register allocator hint).